modules/cmdb/security/adaptor.go

```diff
@@ -0,0 +1,178 @@
+		if err != nil {
+			return false, err
+		}
+		if len(member) > 0 {
+			return true, nil
+		}
+		org, err := this.Db.GetOrg(project.OrgID)
+		if err != nil {
+			return false, err
+		}
+		return org.IsPublic, nil
+	}
+	return true, nil
+}
+
+func (this *PermissionAdaptor) SetCheckRequest(ctx context.Context, req apistructs.PermissionCheckRequest) context.Context {
+	return context.WithValue(ctx, "checkRequest", req)
+}
+
+func (this *PermissionAdaptor) GetCheckRequest(ctx context.Context) apistructs.PermissionCheckRequest {
+	return ctx.Value("checkRequest").(apistructs.PermissionCheckRequest)
+}
+
+func (this *PermissionAdaptor) SetUserID(ctx context.Context, userID string) context.Context {
+	return context.WithValue(ctx, "userID", userID)
+}
+
+func (this *PermissionAdaptor) GetUserID(ctx context.Context) string {
+	return ctx.Value("userID").(string)
+}
+
+func (this *PermissionAdaptor) SetScopeType(ctx context.Context, ScopeType apistructs.ScopeType) context.Context {
+	return context.WithValue(ctx, "ScopeType", ScopeType)
+}
+
+func (this *PermissionAdaptor) GetScopeType(ctx context.Context) apistructs.ScopeType {
+	return ctx.Value("ScopeType").(apistructs.ScopeType)
+}
+
+func (this *PermissionAdaptor) SetScopeID(ctx context.Context, ScopeID int64) context.Context {
+	return context.WithValue(ctx, "ScopeID", ScopeID)
+}
+
+func (this *PermissionAdaptor) GetScopeID(ctx context.Context) int64 {
+	return ctx.Value("ScopeID").(int64)
+}
+
+func (this *PermissionAdaptor) RegisterHandler(handler PermissionHandler) {
+	this.Handlers = append(this.Handlers, handler)
+}
+
+func (this *PermissionAdaptor) CheckPermission(req apistructs.PermissionCheckRequest) (bool, error) {
+	var ctx = context.Background()
+
+	ctx = this.SetScopeID(ctx, int64(req.ScopeID))
+	ctx = this.SetScopeType(ctx, req.Scope)
+	ctx = this.SetUserID(ctx, req.UserID)
+	ctx = this.SetCheckRequest(ctx, req)
+
+	for _, handler := range this.Handlers {
+		// todo multiple handler maybe access true
+		if handler.Access(ctx) {
+			return handler.CheckProcessHandler(ctx).check(ctx)
+		}
+	}
+
+	return false, fmt.Errorf("not find user handler")
+}
+
+func (this *PermissionAdaptor) PermissionList(userID string, scopeType apistructs.ScopeType, scopeID int64) (apistructs.PermissionList, error) {
+	var ctx = context.Background()
+
+	ctx = this.SetScopeID(ctx, scopeID)
+	ctx = this.SetScopeType(ctx, scopeType)
+	ctx = this.SetUserID(ctx, userID)
+
+	for _, handler := range this.Handlers {
+		// todo multiple handler maybe access true
+		if handler.Access(ctx) {
+			list, err := handler.PermissionListProcessHandler(ctx).permissionList(ctx)
+			if list == nil || err != nil {
+				return apistructs.PermissionList{
+					Access:           false,
+					PermissionList:   make([]apistructs.ScopeResource, 0),
+					ResourceRoleList: make([]apistructs.ScopeResource, 0),
+					Roles:            make([]string, 0),
+					Exist:            true,
+				}, err
+			}
+
+			return *list, err
```

Comment:
get all permission differ than check permission.

it should not just return only one handler accessed, it maybe accessed as low-permission role, we shall combine all accessed handler's role/permission.

Reply:
if list is empty or has error, i think it should return empty permission, like old logic